src/main/java/seedu/address/model/contact/Contact.java

```diff
@@ -61,16 +65,19 @@ public Address getAddress() {
     }
 
     /**
-     * Returns true if both persons have the same name.
+     * Returns true if both persons of the same name have at least one other identity field that is the same.
      * This defines a weaker notion of equality between two persons.
      */
-    public boolean isSamePerson(Person otherPerson) {
-        if (otherPerson == this) {
+    public boolean isSameContact(Contact otherContact) {
+        if (otherContact == this) {
             return true;
         }
 
-        return otherPerson != null
-                && otherPerson.getName().equals(getName());
+        return otherContact != null
+                && otherContact.getName().equals(getName())
+                && otherContact.getEmail().equals(getEmail())
+                && otherContact.getPhone().equals(getPhone())
+                && otherContact.getTelegram().equals(getTelegram());
     }
```

Comment:
The stated behaviour in the javadoc comments and the actual behaviour do not match. The method as it is written is actually checking if *all* the fields are the same. It should be something like 
```java
        return otherContact != null
                && otherContact.getName().equals(getName())
                && (otherContact.getEmail().equals(getEmail())
                         || otherContact.getPhone().equals(getPhone())
                         || otherContact.getTelegram().equals(getTelegram()));
```
That being said, good catch on enforcing this new notion of same contact! It definitely makes more sense than not allowing 2 people with the same name.

Just remember to update the `isSamePerson()` in `PersonTest.java` (lines 36-39) eg. by removing one of the `withXYZ(...)` so that the CI can pass.